js/components/frame.js

```diff
@@ -557,6 +558,8 @@ class Frame extends ImmutableComponent {
     this.webview.addEventListener('content-blocked', (e) => {
       if (e.details[0] === 'javascript') {
         windowActions.setBlockedBy(this.frame, 'noScript', e.details[1])
+      } else if (!this.allowRunningInsecureContent()) {
```

Comment:
we can now manage running and displaying insecure content through the `displayInsecureContent` and `runInsecureContent` content settings which will allow us to treat them like any other site settings. I know you already did a lot of work using the old method and I'm sorry I didn't catch this sooner, but setting these through the webview is problematic because it restarts the renderer process and will potentially break some sites. We had a small number of exceptions before, but if we're opening this up as a general setting we should transition to the contentSetting